### ISSUE TITLE
Ozark 1.0.0-m4 does not work on wildfly

### ISSUE THREAD

**Issue by gtudan** [mvc-spec/ozark#206]
It looks like the current master somehow broke on wildfly. I tried with some of the example apps and they all show the default jax-rs behavios (rendering 204 for controllers with view-annotations or "hello.jsp" for the "return hello.jsp"-pattern). The " Ozark version 1.0.0-SNAPSHOT started"-messages show up in the log and there aren't any exceptions.

I tried wildfly 11 to 14 - same results.

Here's the log from wildfly 14.

``` 
=========================================================================

  JBoss Bootstrap Environment

  JBOSS_HOME: /home/gtudan/local/wildfly-14.0.1.Final

  JAVA: /usr/java/jdk1.8.0_181-amd64//bin/java

  JAVA_OPTS:  -server -Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Djboss.modules.system.pkgs=org.jboss.byteman -Djava.awt.headless=true 

=========================================================================

16:19:55,503 INFO  [org.jboss.modules] (main) JBoss Modules version 1.8.6.Final
16:19:55,893 INFO  [org.jboss.msc] (main) JBoss MSC version 1.4.3.Final
16:19:55,901 INFO  [org.jboss.threads] (main) JBoss Threads version 2.3.2.Final
16:19:55,997 INFO  [org.jboss.as] (MSC service thread 1-2) WFLYSRV0049: WildFly Full 14.0.1.Final (WildFly Core 6.0.2.Final) starting
16:19:56,582 INFO  [org.wildfly.security] (ServerService Thread Pool -- 20) ELY00001: WildFly Elytron version 1.6.0.Final
16:19:56,961 INFO  [org.jboss.as.controller.management-deprecated] (Controller Boot Thread) WFLYCTL0028: Attribute 'security-realm' in the resource at address '/core-service=management/management-interface=http-interface' is deprecated, and may be removed in a future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation.
16:19:57,032 INFO  [org.jboss.as.controller.management-deprecated] (ServerService Thread Pool -- 34) WFLYCTL0028: Attribute 'security-realm' in the resource at address '/subsystem=undertow/server=default-server/https-listener=https' is deprecated, and may be removed in a future version. See the attribute description in the output of the read-resource-description operation to learn more about the deprecation.
16:19:57,056 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0039: Creating http management service using socket-binding (management-http)
16:19:57,070 INFO  [org.xnio] (MSC service thread 1-6) XNIO version 3.6.5.Final
16:19:57,079 INFO  [org.xnio.nio] (MSC service thread 1-6) XNIO NIO Implementation Version 3.6.5.Final
16:19:57,107 INFO  [org.jboss.remoting] (MSC service thread 1-1) JBoss Remoting version 5.0.8.Final
16:19:57,154 INFO  [org.jboss.as.connector.subsystems.datasources] (ServerService Thread Pool -- 41) WFLYJCA0004: Deploying JDBC-compliant driver class org.h2.Driver (version 1.4)
16:19:57,190 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 48) WFLYCLINF0001: Activating Infinispan subsystem.
16:19:57,200 INFO  [org.wildfly.extension.io] (ServerService Thread Pool -- 49) WFLYIO001: Worker 'default' has auto-configured to 8 core threads with 64 task threads based on your 4 available processors
16:19:57,205 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-6) WFLYJCA0010: Unbound data source [java:jboss/datasources/ExampleDS]
16:19:57,233 INFO  [org.jboss.as.jaxrs] (ServerService Thread Pool -- 50) WFLYRS0016: RESTEasy version 3.6.1.Final
16:19:57,239 INFO  [org.jboss.as.jsf] (ServerService Thread Pool -- 55) WFLYJSF0007: Activated the following JSF Implementations: [main]
16:19:57,250 INFO  [org.wildfly.extension.microprofile.config.smallrye._private] (ServerService Thread Pool -- 57) WFLYCONF0001: Activating WildFly MicroProfile Config Subsystem
16:19:57,257 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-2) WFLYJCA0018: Started Driver service with driver-name = h2
16:19:57,257 INFO  [org.jboss.as.connector] (MSC service thread 1-2) WFLYJCA0009: Starting JCA Subsystem (WildFly/IronJacamar 1.4.11.Final)
16:19:57,255 INFO  [org.jboss.as.mail.extension] (MSC service thread 1-5) WFLYMAIL0002: Unbound mail session [java:jboss/mail/Default]
16:19:57,263 INFO  [org.wildfly.extension.microprofile.opentracing] (ServerService Thread Pool -- 59) WFLYTRACEXT0001: Activating MicroProfile OpenTracing Subsystem
16:19:57,270 INFO  [org.wildfly.extension.microprofile.health.smallrye] (ServerService Thread Pool -- 58) WFLYHEALTH0001: Activating Eclipse MicroProfile Health Subsystem
16:19:57,310 INFO  [org.jboss.as.naming] (ServerService Thread Pool -- 60) WFLYNAM0001: Activating Naming Subsystem
16:19:57,328 INFO  [org.jboss.as.mail.extension] (MSC service thread 1-8) WFLYMAIL0001: Bound mail session [java:jboss/mail/Default]
16:19:57,328 INFO  [org.jboss.as.naming] (MSC service thread 1-1) WFLYNAM0003: Starting Naming Service
16:19:57,331 WARN  [org.jboss.as.txn] (ServerService Thread Pool -- 68) WFLYTX0013: The node-identifier attribute on the /subsystem=transactions is set to the default value. This is a danger for environments running multiple servers. Please make sure the attribute value is unique.
16:19:57,338 INFO  [org.jboss.as.security] (ServerService Thread Pool -- 66) WFLYSEC0002: Activating Security Subsystem
16:19:57,359 INFO  [org.jboss.as.security] (MSC service thread 1-2) WFLYSEC0001: Current PicketBox version=5.0.3.Final
16:19:57,361 INFO  [org.jboss.as.ejb3] (MSC service thread 1-1) WFLYEJB0481: Strict pool slsb-strict-max-pool is using a max instance size of 64 (per class), which is derived from thread worker pool sizing.
16:19:57,375 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-4) WFLYUT0003: Undertow 2.0.13.Final starting
16:19:57,365 INFO  [org.jboss.as.ejb3] (MSC service thread 1-7) WFLYEJB0482: Strict pool mdb-strict-max-pool is using a max instance size of 16 (per class), which is derived from the number of CPUs on this host.
16:19:57,387 INFO  [org.jboss.as.webservices] (ServerService Thread Pool -- 70) WFLYWS0002: Activating WebServices Extension
16:19:57,475 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 69) WFLYUT0014: Creating file handler for path '/home/gtudan/local/wildfly-14.0.1.Final/welcome-content' with options [directory-listing: 'false', follow-symlink: 'false', case-sensitive: 'true', safe-symlink-paths: '[]']
16:19:57,495 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-1) WFLYUT0012: Started server default-server.
16:19:57,496 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-8) WFLYUT0018: Host default-host starting
16:19:57,680 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-8) WFLYUT0006: Undertow HTTP listener default listening on 127.0.0.1:8080
16:19:57,735 INFO  [org.jboss.as.ejb3] (MSC service thread 1-1) WFLYEJB0493: EJB subsystem suspension complete
16:19:57,794 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-6) WFLYJCA0001: Bound data source [java:jboss/datasources/ExampleDS]
16:19:57,862 INFO  [org.jboss.as.patching] (MSC service thread 1-6) WFLYPAT0050: WildFly Full cumulative patch ID is: base, one-off patches include: none
16:19:57,895 INFO  [org.jboss.as.server.deployment.scanner] (MSC service thread 1-3) WFLYDS0013: Started FileSystemDeploymentService for directory /home/gtudan/local/wildfly-14.0.1.Final/standalone/deployments
16:19:57,900 WARN  [org.jboss.as.domain.management.security] (MSC service thread 1-1) WFLYDM0111: Keystore /home/gtudan/local/wildfly-14.0.1.Final/standalone/configuration/application.keystore not found, it will be auto generated on first use with a self signed certificate for host localhost
16:19:57,905 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0027: Starting deployment of "test-handlebars.war" (runtime-name: "test-handlebars.war")
16:19:58,024 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-7) WFLYUT0006: Undertow HTTPS listener https listening on 127.0.0.1:8443
16:19:58,092 INFO  [org.jboss.ws.common.management] (MSC service thread 1-6) JBWS022052: Starting JBossWS 5.2.3.Final (Apache CXF 3.2.5.jbossorg-1) 
16:19:58,532 INFO  [org.jboss.weld.deployer] (MSC service thread 1-5) WFLYWELD0003: Processing weld deployment test-handlebars.war
16:19:58,593 INFO  [org.hibernate.validator.internal.util.Version] (MSC service thread 1-5) HV000001: Hibernate Validator 6.0.13.Final
16:19:58,843 INFO  [org.jboss.weld.Version] (MSC service thread 1-7) WELD-000900: 3.0.5 (Final)
16:19:59,010 INFO  [org.infinispan.factories.GlobalComponentRegistry] (MSC service thread 1-6) ISPN000128: Infinispan version: Infinispan 'Estrella Galicia' 9.3.1.Final
16:19:59,181 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 72) WFLYCLINF0002: Started client-mappings cache from ejb container
16:19:59,583 INFO  [org.mvcspec.ozark.servlet.OzarkContainerInitializer] (ServerService Thread Pool -- 74) Ozark version 1.0.0-SNAPSHOT started
16:19:59,600 WARN  [io.jaegertracing.internal.senders.SenderResolver] (ServerService Thread Pool -- 74) No suitable sender found. Using NoopSender, meaning that data will not be sent anywhere!
16:19:59,629 INFO  [io.jaegertracing.Configuration] (ServerService Thread Pool -- 74) Initialized tracer=JaegerTracer(version=Java-0.30.6, serviceName=test-handlebars.war, reporter=RemoteReporter(sender=NoopSender(), closeEnqueueTimeout=1000), sampler=RemoteControlledSampler(maxOperations=2000, manager=HttpSamplingManager(hostPort=localhost:5778), sampler=ProbabilisticSampler(tags={sampler.type=probabilistic, sampler.param=0.001})), tags={hostname=G-PC.local, jaeger.version=Java-0.30.6, ip=192.168.39.1}, zipkinSharedRpcSpan=false, expandExceptionLogs=false)
16:19:59,862 INFO  [org.jboss.resteasy.resteasy_jaxrs.i18n] (ServerService Thread Pool -- 74) RESTEASY002225: Deploying javax.ws.rs.core.Application: class org.mvcspec.ozark.test.handlebars.MyApplication$Proxy$_$$_WeldClientProxy
16:19:59,865 INFO  [org.jboss.resteasy.resteasy_jaxrs.i18n] (ServerService Thread Pool -- 74) RESTEASY002200: Adding class resource org.mvcspec.ozark.test.handlebars.PersonController from Application class org.mvcspec.ozark.test.handlebars.MyApplication$Proxy$_$$_WeldClientProxy
16:19:59,896 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 74) WFLYUT0021: Registered web context: '/test-handlebars' for server 'default-server'
16:19:59,905 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0010: Deployed "test-handlebars.war" (runtime-name : "test-handlebars.war")
16:19:59,943 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0212: Resuming server
16:19:59,945 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://127.0.0.1:9990/management
16:19:59,945 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0051: Admin console listening on http://127.0.0.1:9990
16:19:59,945 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly Full 14.0.1.Final (WildFly Core 6.0.2.Final) started in 4698ms - Started 433 of 616 services (326 services are lazy, passive or on-demand)
```

**Comment by gtudan** [comment]
Excuse me - I somewhat forgot that the examples didn't ship the ozark-resteasy-module out-of-the-box. That's what you get for being on holiday for to long ;-) 

Still there's something strange about the current master with wildfly that led me to try the example apps. I'm getting different errors on different wildfly versions. 14 compains that it cannot find an engine for the Viewable (with jsp), 13 gave me stackoverflow exceptions in weld.

I'll try to investigate and file a proper issue once I can put my finger on it.

**Comment by chkal** [comment]
Ah, ok, thanks! I started to get worried, as we just released `1.0.0-m4`. :laughing:
